packages/transaction-controller/src/TransactionController.ts

```diff
@@ -302,7 +524,7 @@ export class TransactionController extends BaseControllerV1<
       error: normalizeTxError(error),
       status: TransactionStatus.failed,
     };
-    this.hub.emit('transaction-failed', {
+    this.messagingSystem.publish(`${controllerName}:transactionFailed`, {
```

Comment:
Is this is a formal standard now?

I worry this could make handling of basic events more complex since now every entity that wants to listen needs to have a messenger object to subscribe to?

Furthermore, if the scope of this PR is to update the `BaseController`, I'd feel much more comfortable doing only the minimum refactors needed to achieve that, with any further required changes being the result of specific discussion and analysis.

Naturally the `TransactionController` is very high risk and has very recently undergone a large refactor due to the migration so stability is paramount.